### ISSUE TITLE
[Serverless]: Inference management and ML

### ISSUE THREAD

**Issue by serenachou** [elastic/docs-content#156]
### Serverless Docs

Welcome to Elastic Serverless

### Description

Missing guidance on ML / inference capabilities in the Serverless documentation. Would like to see better guidance on managing Inference endpoints, and using them in retrievers and for semantic reranking and with semantic_text

### Resources and additional context

For GA we want to ensure customers know what we're capable of as a distributed multi cloud vector database.

**Comment by kosabogi** [comment]
The documentation team will be making a decision this week on our approach to handling cases where documentation would potentially be duplicated on serverless. We are considering whether to maintain deployment-specific guides or consolidate content to avoid redundancy. Once a decision is made, I'll proceed with the necessary updates and move this issue forward accordingly.

**Comment by kosabogi** [comment]
To avoid duplication, we decided to remove redundant content from the Serverless documentation and add a page with links and descriptions for existing tutorials in the Elasticsearch documentation.

After restructuring, the Semantic search page now has the following structure:

- Semantic search: Introduction, description, a list of recommended workflows, and a link to the guide in the Elasticsearch documentation.
- Semantic search with the inference API workflow: Introduction, description, and a link to the guide in the Elasticsearch documentation.
- Semantic search with semantic_text workflow: Introduction, description, and a link to the guide in the Elasticsearch documentation.
- Semantic search with a deployed model workflow: Introduction, description, and a link to the guide in the Elasticsearch documentation.

Updated documentation: [Semantic search](https://www.elastic.co/guide/en/serverless/current/elasticsearch-reference-semantic-search.html)
